File: LeetCode_Algorithm/191(Number_of_1_Bits)/191.py


class Solution:
    def hammingWeight(self, n: int) -> int:
        # sol1 python 29~38ms 91~66%
        # return bin(n).count('1')

        # sol2 bit mask 31~65ms 89~7%
        cnt = 0
        for mask in range(32):
            if (1 << mask) & n:
                cnt += 1
        return (cnt)

print(Solution().hammingWeight(7))

File: LeetCode_Algorithm/191(Number_of_1_Bits)/191.js
var hammingWeight = function(n) {
  // sol2 105ms 40%
  let ans = 0;
  for (let i=0; i < 32; i++) {
    ans += n & 1
    n >>= 1
  } 
  return ans
};

console.log(hammingWeight("00000000000000000000000000001011"))
console.log(hammingWeight("11111111111111111111111111111101"))
